src/NuGet.Core/NuGet.Protocol/RemoteRepositories/HttpFileSystemBasedFindPackageByIdResource.cs

```diff
@@ -297,6 +299,110 @@ public class HttpFileSystemBasedFindPackageByIdResource : FindPackageByIdResourc
             }
         }
 
+#pragma warning disable RS0016 // Add public types and members to the declared API
+        public async Task<(bool success, string contentUri, PackageIdentity packageIdentity)> GetNupkgPackageContentUriAsync(
```

Comment:
Of course, this is a workaround to build. All `FindPackageByIdResourc` should have this method and push up the interface.

Reply:
If we are worried about breaking change, we can make FindPackageByIdResourc2